include/deal.II/base/point.h

```diff
@@ -25,28 +25,46 @@
+ * points in spaces of a priori fixed dimension: rather than using functions
  * like <tt>double f(double x)</tt> and <tt>double f(double x, double y)</tt>,
- * you use <tt>double f(Point<dim> &p)</tt>.
+ * you should use <tt>double f(Point<dim> &p)</tt> instead as it allows writing
+ * dimension independent code.
  *
- * <tt>Point</tt> also serves as a starting point for the implementation of
- * the geometrical primitives like cells, edges, or faces.
  *
- * Within deal.II, we use the <tt>Point</tt> class mainly to denote the points
- * that make up geometric objects. As such, they have a small number of
- * additional operations over general tensors of rank 1 for which we use the
- * <tt>Tensor<1,dim></tt> class. In particular, there is a distance() function
- * to compute the Euclidean distance between two points in space.
+ * <h3>What's a <code>Point@<dim@></code> and what is a <code>Tensor@<1,dim@></code>?</h3>
  *
- * The <tt>Point</tt> class is really only used where the coordinates of an
- * object can be thought to possess the dimension of a length. For all other
- * uses, such as the gradient of a scalar function (which is a tensor of rank
- * 1, or vector, with as many elements as a point object, but with different
- * physical units), we use the <tt>Tensor<1,dim></tt> class.
+ * The Point class is derived from Tensor@<1,dim@> and consequently
+ * shares the latter's member functions and other attributes. In fact,
+ * it has relatively few additional functions itself (the most notable
+ * exception being the distance() function to compute the Euclidean
+ * distance between two points in space), and these two classes can
+ * therefore often be used interchangeably.
+ *
+ * Nonetheless, there are semantic differences that make us use these
+ * classes in different and well-defined contexts. Within deal.II, we
+ * use the <tt>Point</tt> class to denote points in space, i.e., for
+ * vectors (rank-1 tensors) that are <em>anchored at the
+ * origin</em>. On the other hand, vectors that are anchored elsewhere
+ * (and consequently do not represent <em>points</em> in the common
+ * usage of the word) are represented by objects of type
+ * Tensor@<1,dim@>. In particular, this is the case for direction
+ * vectors, normal vectors, gradients, and the differences between two
+ * points (i.e., what you get when you subtract one point from
+ * another): all of these are represented by Tensor@<1,dim@> objects
+ * rather than Point@<dim@>.
+ *
+ * Furthermore, the Point class is only used where the coordinates of
+ * an object can be thought to possess the dimension of a length. An
+ * object that represents the weight, height, and cost of an object is
+ * neither a point nor a tensor (because it lacks the transformation
+ * properties under rotation of the coordinate system) and should
+ * consequently not be represented by either of these classes. Use an
+ * array of size 3 in this case, or the <code>std_cxx11::array</code>
+ * class.
```

Comment:
I'm not quite sure I follow. If we do do this, then we shouldn't -- the example given is simply a collection of 3 numbers. I think one could use a `std::vector` in these cases, but that's also questionable.